sched/sched.h

```diff
@@ -17,6 +17,8 @@
 
 #include "return.h"
 
+#define RET_CHECK(ret) if(ret != RET_SUCCESS) return ret;
```

Comment:
macros probably shouldn't go in this file, if you want to add a macro add it in the sched/macros directory. I would also use this macro with caution as a lot of times there needs to be some kind of cleanup before a return and implicitly returning without saving position could cause some bugs if the user doesnt know